pkg/storage/stores/shipper/bloomshipper/store.go

```diff
@@ -0,0 +1,105 @@
+package bloomshipper
+
+import (
+	"context"
+	"time"
+
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+
+	"github.com/grafana/loki/pkg/logproto"
+	"github.com/grafana/loki/pkg/storage"
+	"github.com/grafana/loki/pkg/storage/config"
+)
+
+// TODO(chaudum): This is just a placeholder and needs to be replaced by actual
+// bloom block querier interface.
+type BlockQuerier interface {
+}
+
+type ForEachBlockCallback func(bq BlockQuerier) error
+
+type ReadShipper interface {
+	ForEachBlock(ctx context.Context, tenant string, from, through time.Time, fingerprints []uint64, callback ForEachBlockCallback) error
+}
+
+type WriteShipper interface {
+}
+
+type Shipper interface {
+	ReadShipper
+	WriteShipper
```

Comment:
Does this block also have bloom-compactor in mind? otherwise why is WriteShipper needed?

Reply:
I could remove it, because it's not needed for the purposes of this PR.